javascript/javascript3/week1/giphy.js

```diff
@@ -0,0 +1,39 @@
+const apiKey = "KzWbfntmytNcoeo9ygrJJdIstwy3bjtB";
+const giphyInput = document.getElementById("giphyInput");
+const giphyInputNumber = document.getElementById("giphyInputNumber");
+const searchGif = document.getElementById("searchGif");
+const imageResult = document.getElementById("imageResult");
```

Comment:
what do you mean here?thanks

Reply:
Try to break the large function into several reusable ones to make the code become lighter and nicer.Can you show me where should I put the function out?thanks

Reply:
Think of a function that takes input of a string(i.e id) and outputs document.getElementById({that_id_you_passed}). Does it not work the same way as here? Whenever you are writing same thing for more than twice, you need to consider function and reusability. It would increase the number of lines, but will produce better quality code. Ping me if it doesn't make sense to you.

Reply:
Default is not Chengdu, I just make some info inside my HTML. There is one button for getting the current city weather info and another input is for searching the city info.I just want to make the website looks better, so I left some info before we searched in HTML.

Reply:
Okay, got that. But try to make it more universal and reachable. You may follow your phone weather app and take something interesting and implement on your app. This is a super nice app where you can implement so many things and impress your future employer.

Reply:
function idGet(idName){
    return  `const ${idName} = document.getElementById("${idName}")`

}
idGet(giphyInput); 

Do you mean this ?

Reply:
Yes, perfect. only thing I would change here is the naming. Function names should always describe what it does and start with a verb, getTheElement, calculateTheAge, countRows so on.

Reply:
oh, yes you are right. Thanks for telling. I learned a lot from you.